platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -35,6 +35,99 @@ The previous step will help you identify potential problems by issuing deprecati
 . Run `gradle wrapper --gradle-version {gradleVersion}` to update the project to {gradleVersion}.
 . Try to run the project and debug any errors using the <<troubleshooting.adoc#troubleshooting, Troubleshooting Guide>>.
 
+[[changes_8.7]]
+== Upgrading from 8.6 and earlier
+
+=== Deprecations
+
+[[deprecate_self_resolving_dependency]]
+==== Deprecated `SelfResolvingDependency`
+
+The `SelfResolvingDependency` interface has been deprecated for removal in Gradle 9.0.
+This type dates back to the first versions of Gradle, where some dependencies could be resolved independently.
+Now, all dependencies should be resolved as part of a dependency graph using a `Configuration`.
+
+Currently, `ProjectDependency` and `FileCollectionDependency` implement this interface.
+In Gradle 9.0, these types will no longer implement `SelfResolvingDependency`.
+Instead, they will both directly implement `Dependency`.
+
+As such, the following methods of `ProjectDependency` and `FileCollectionDependency` will no longer be available:
+
+- `resolve`
+- `resolve(boolean)`
+- `getBuildDependencies`
+
+Consider the following scripts that showcase the deprecated interface and its replacement:
+
+=====
+[.multi-language-sample]
+======
+.build.gradle.kts
+[source,kotlin]
+----
+plugins {
+    id("java-library")
+}
+
+dependencies {
+    implementation(files("bar.txt"))
+    implementation(project(":foo"))
+}
+
+tasks.register("resolve") {
+    val files = configurations["runtimeClasspath"].files
+    doLast {
+        // Deprecated
+        configurations["implementation"].dependencies.withType<FileCollectionDependency>() {
+            resolve()
+            resolve(true)
+        }
+        configurations["implementation"].dependencies.withType<ProjectDependency>() {
+            resolve()
+            resolve(true)
+        }
+
+        // Resolve a Configuration instead
```

Comment:
🤔 I think this would be clearer with an additional comment like "iterating through the files implicitly resolves the dependencies.

I also really think this should be split into 2 tasks "resolveDeprecated" and "resolve" (showing the replacement) that both do something with the resolution result..  This doesn't show the equivalent operation (iterating the file names) using the old way and new way.

Reply:
Added asserts in the first (deprecated) example to show these methods resolve to files and split this up into two tasks